api/app/utils/auth.ts

```diff
@@ -0,0 +1,29 @@
+import jwt from "jsonwebtoken";
+
+// Verify the OAuth token with GitHub API
+export async function verifyGitHubToken(token: string) {
+    // Use your GitHub API endpoint and headers
+    const response = await fetch("https://api.github.com/user", {
+      headers: {
+        Authorization: `Bearer ${process.env.GITHUB_OAUTH_TOKEN}`,
+        Accept: "application/vnd.github.v3+json",
+      },
+    });
+  
+    if (!response.ok) {
+      throw new Error("Failed to verify GitHub token");
+    }
+  
+    return response.json();
+  }
+  
+  // Generate JWT with user information
+  export function generateJwtToken(user: any) {
+    const secretKey = `process.env.JWT_SECRET_KEY`;
+    const token = jwt.sign(
+      { userId: user.id, username: user.githubUsername, email: user.email, permissions: user.permissions, githubAuthToken: user.authToken },
+      secretKey,
+      { expiresIn: "1h" }
```

Comment:
We have an hardcoded expiry time here, is that fine?

Also, I think it should be synced to the github OAuth token. That is next-auth is in control of new OAuth tokens which in turn controls generation of JWT. 
Handling refresh token with next-auth might get convoluted

Reply:
The time is for testing, we can move it to a constant in the last PR. 

On the second point are you saying we should remove the expiresIn on the backend or should we sync the time the one provided in the next OAuth?

Reply:
For more context, we can get user permission level from the githubAuthToken data. So we can encode that with the jwt and the githubAuthToken, or we can do that with middleware on the routes for all permission levels.